### PR TITLE
boards:shields: Round Display for Xiao

### PR DESCRIPTION
**Pull request by nicogou**
Added the Round Display for Xiao boards as a shield.
Removed the required: true from the reset-gpios property of the existing gcx01x driver as it is not routed on the board.
Also added the necessary conf file in the LVGL subsys sample for Xiao BLE (Sense) boards.

### CODE REVIEW

File: samples/subsys/display/lvgl/boards/xiao_ble_nrf52840.conf [hunk 1]
```diff
@@ -0,0 +1 @@
+CONFIG_INPUT=y
```
Comment (danieldegrasse) [inline samples/subsys/display/lvgl/boards/xiao_ble_nrf52840.conf:1]: Why does this need to be set here, versus being enabled in the shield's `Kconfig.defconfig`?

File: dts/bindings/display/galaxycore,gc9x01x.yaml [hunk 2]
```diff
@@ -34,7 +34,6 @@ include: [spi-device.yaml, display-controller.yaml, lcd-controller.yaml]
 properties:
   reset-gpios:
     type: phandle-array
-    required: true
```
Comment (danieldegrasse) [inline dts/bindings/display/galaxycore,gc9x01x.yaml:37]: This change is also covered by #70683, you can likely drop it here

File: boards/shields/seeed_xiao_round_display/doc/img/rounddisplay.jpg
Comment (danieldegrasse) [inline boards/shields/seeed_xiao_round_display/doc/img/rounddisplay.jpg:1]: nit- can you provide a more descriptive name for this file? Maybe `seed_xiao_round_display.jpg`?

File: samples/subsys/display/lvgl/boards/xiao_ble_nrf52840.conf (hunk 1 above)
Comment (nicogou) [reply samples/subsys/display/lvgl/boards/xiao_ble_nrf52840.conf:1]: My reasoning was that if it's enabled in the shield's `Kconfig.defconfig`, and the shield is used only for display and not using the input then more files than needed would be compiled. I thought I'd try to avoid that, but I can move it over there if that's not an issue.

Comment (danieldegrasse) [reply samples/subsys/display/lvgl/boards/xiao_ble_nrf52840.conf:1]: That does make sense. Typically in the `Kconfig.defconfig` for shields I have enabled `CONFIG_INPUT` when `CONFIG_LVGL` is set. Of course you can use the input subsystem without LVGL, but I would imagine most users of LVGL do want touch support if their display offers it.

Example: https://github.com/zephyrproject-rtos/zephyr/blob/89493ee56b6cbd51b1117796b5dc2a2666bc22e8/boards/shields/g1120b0mipi/Kconfig.defconfig#L1

Comment (nicogou) [reply samples/subsys/display/lvgl/boards/xiao_ble_nrf52840.conf:1]: Fair enough, I'll do that

Comment (nicogou) [reply samples/subsys/display/lvgl/boards/xiao_ble_nrf52840.conf:1]: Well, I thought I'd do the changes quick and easy, but something must have happened in the 250+ commits that happened between last time I looked at this and tonight, because after rebasing it's not working anymore...
I'll try to look into it some more later this week

File: boards/shields/seeed_xiao_round_display/doc/img/rounddisplay.jpg
Comment (kartben) [reply boards/shields/seeed_xiao_round_display/doc/img/rounddisplay.jpg:1]: What Daniel said - also would be great to use webp format. Converting with e.g https://developers.google.com/speed/webp/docs/cwebp get the file size down to 28KB.

File: samples/subsys/display/lvgl/README.rst [hunk 3]
```diff
@@ -41,6 +41,7 @@ for Arduino connectors, for example:
 - :ref:`adafruit_2_8_tft_touch_v2` and :ref:`nrf52840dk_nrf52840`
 - :ref:`buydisplay_2_8_tft_touch_arduino` and :ref:`nrf52840dk_nrf52840`
 - :ref:`ssd1306_128_shield` and :ref:`frdm_k64f`
+- :ref:`seeed_xiao_round_display` and :ref:`xiao_ble`
```
Comment (kartben) [inline samples/subsys/display/lvgl/README.rst:44]: the statement at lines 38-39 is now wrong since it's more than Arduino connectors, and now e.g. XIAO connector. Maybe rephrase as: "Display shield and a board which provides a configuration for corresponding connector, for example:"

File: boards/shields/seeed_xiao_round_display/seeed_xiao_round_display.overlay [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    };
+
+    aliases {
+        rtc = &pcf8563_xiao_round_display;
+    };
+};
+
+&xiao_serial {
+    status = "disabled";
+};
```
Comment (kartben) [inline boards/shields/seeed_xiao_round_display/seeed_xiao_round_display.overlay:31]: would be nice to have a comment to explain why it's there (or maybe it's not needed and should be dropped?)

**Review by kartben — CHANGES_REQUESTED** [review]

File: boards/shields/seeed_xiao_round_display/seeed_xiao_round_display.overlay (hunk 4 above)
Comment (nicogou) [reply boards/shields/seeed_xiao_round_display/seeed_xiao_round_display.overlay:35]: The pins are used by the display and touch controller. I added a comment explaining that in the overlay.

File: samples/subsys/display/lvgl/boards/xiao_ble_nrf52840.conf (hunk 1 above)
Comment (nicogou) [reply samples/subsys/display/lvgl/boards/xiao_ble_nrf52840.conf:1]: Changes made, I also fixed a missing display-inversion property in the gc9a01 node causing wrong colors to be displayed.

File: boards/shields/seeed_xiao_round_display/Kconfig.defconfig [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+endchoice
+
+config LV_COLOR_16_SWAP
+	default y
+
+endif # LVGL
+endif # DISPLAY
+
+if INPUT
+config INPUT_CHSC6X
```
Comment (danieldegrasse) [inline boards/shields/seeed_xiao_round_display/Kconfig.defconfig:24]: You shouldn't need to enable this driver here- the Kconfig in the input subsystem should be enabled automatically by the `"chipsemi,chsc6x"` compatible in your devicetree

File: boards/shields/seeed_xiao_round_display/doc/index.rst [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+XIAO development boards. It features a fully covered touch screen on one side,
+designed as a 39mm disc. It contains onboard RTC, charge chip, TF card slot
+within its compact size, perfect for interactive displays in smart home,
+wearables and more.
+
+More information can be found on `the Getting Started page`_
+
+.. figure:: img/seeed_xiao_round_display.webp
+     :align: center
+     :alt: Seeed Studio XIAO Round Display
```
Comment (kartben) [inline boards/shields/seeed_xiao_round_display/doc/index.rst:19]: Image is taking a lot of space on the page :) (see https://builds.zephyrproject.io/zephyr/pr/71239/docs/boards/shields/seeed_xiao_round_display/doc/index.html) 

```suggestion
     :width: 450 px
     :alt: Seeed Studio XIAO Round Display
```

alternatively, resize the source image so that it's "natively" smaller

**Review by kartben — CHANGES_REQUESTED** [review]
one last comment from me regarding docs, but looks good otherwise, thanks!

File: boards/shields/seeed_xiao_round_display/Kconfig.defconfig (hunk 5 above)
Comment (nicogou) [reply boards/shields/seeed_xiao_round_display/Kconfig.defconfig]: Ah yup, just did that.

File: boards/shields/seeed_xiao_round_display/doc/index.rst (hunk 6 above)
Comment (nicogou) [reply boards/shields/seeed_xiao_round_display/doc/index.rst:19]: Image resized ;)

File: boards/shields/seeed_xiao_round_display/seeed_xiao_round_display.overlay [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+ */
+
+/ {
+    chosen {
+        zephyr,display = &gc9a01_xiao_round_display;
+    };
+
+    vbatt {
+        compatible = "voltage-divider";
+        io-channels = <&adc 0>;
```
Comment (kartben) [inline boards/shields/seeed_xiao_round_display/seeed_xiao_round_display.overlay:14]: I don't think this works, as this would fail on e.g ESP32S3 XIAO where adc nodes are `adc0` and `adc1`, not `adc`?

@erwango what would be a correct way to do this?

Comment (erwango) [reply boards/shields/seeed_xiao_round_display/seeed_xiao_round_display.overlay:14]: Idea would be to add a `xiao_adc` nodelabel (similarly to existing `xiao_serial` for instance) which would have to be defined by compatible boards.

File: boards/shields/seeed_xiao_round_display/seeed_xiao_round_display.overlay [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+    gc9a01_xiao_round_display: gc9a01@0 {
+        status = "okay";
+        compatible = "galaxycore,gc9x01x";
+        reg = <0>;
+        spi-max-frequency = <100000000>;
+        cmd-data-gpios = <&xiao_d 3 GPIO_ACTIVE_HIGH>;
+        pixel-format = <16>;
+        width = <240>;
+        height = <240>;
+		display-inversion;
```
Comment (erwango) [inline boards/shields/seeed_xiao_round_display/seeed_xiao_round_display.overlay:67]: Nit: indentation

File: boards/shields/seeed_xiao_round_display/doc/index.rst [hunk 9]
```diff
@@ -0,0 +1,40 @@
+.. _seeed_xiao_round_display:
+
+Seeed Studio XIAO Round Display
+#################################
```
Comment (erwango) [inline boards/shields/seeed_xiao_round_display/doc/index.rst:4]:
```suggestion
###############################
```

File: boards/shields/seeed_xiao_round_display/doc/index.rst [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+
+     Seeed Studio XIAO Round Display (Credit: Seeed Studio)
+
+Programming
+***********
+
+Set ``-DSHIELD=seeed_xiao_round_display`` when you invoke ``west build``.
+
+LVGL Basic Sample
+==========================
```
Comment (erwango) [inline boards/shields/seeed_xiao_round_display/doc/index.rst:29]:
```suggestion
===============
```

File: boards/shields/seeed_xiao_round_display/seeed_xiao_round_display.overlay [hunk 11]
```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2024, Nicolas Goualard <nicolas.goualard@sfr.fr>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/ {
```
Comment (mniestroj) [inline boards/shields/seeed_xiao_round_display/seeed_xiao_round_display.overlay:7]:
```suggestion
 */

#include <freq.h>

/ {
```

File: boards/shields/seeed_xiao_round_display/seeed_xiao_round_display.overlay (hunk 8 above)
Comment (mniestroj) [reply boards/shields/seeed_xiao_round_display/seeed_xiao_round_display.overlay:67]: @nicogou Please use tabs in the whole file, so it will be consistent with other Zephyr dts, dtsi and overlay files.

File: boards/shields/seeed_xiao_round_display/seeed_xiao_round_display.overlay [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+
+    sdhc_xiao_round_display: sdhc@1 {
+        compatible = "zephyr,sdhc-spi-slot";
+        reg = <1>;
+        status = "okay";
+        mmc {
+            compatible = "zephyr,sdmmc-disk";
+            status = "okay";
+        };
+        spi-max-frequency = <24000000>;
```
Comment (mniestroj) [inline boards/shields/seeed_xiao_round_display/seeed_xiao_round_display.overlay:78]:
```suggestion
        spi-max-frequency = <DT_FREQ_M(24)>;
```

File: boards/shields/seeed_xiao_round_display/seeed_xiao_round_display.overlay [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+
+&xiao_spi {
+    status = "okay";
+    cs-gpios = <&xiao_d 1 GPIO_ACTIVE_LOW>, <&xiao_d 2 GPIO_ACTIVE_LOW>;
+
+    gc9a01_xiao_round_display: gc9a01@0 {
+        status = "okay";
+        compatible = "galaxycore,gc9x01x";
+        reg = <0>;
+        spi-max-frequency = <100000000>;
```
Comment (mniestroj) [inline boards/shields/seeed_xiao_round_display/seeed_xiao_round_display.overlay:62]:
```suggestion
        spi-max-frequency = <DT_FREQ_M(100)>;
```

File: boards/shields/seeed_xiao_round_display/seeed_xiao_round_display.overlay [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+        cmd-data-gpios = <&xiao_d 3 GPIO_ACTIVE_HIGH>;
+        pixel-format = <16>;
```
Comment (mniestroj) [inline boards/shields/seeed_xiao_round_display/seeed_xiao_round_display.overlay:64]: Please use `PANEL_PIXEL_FORMAT_RGB_*` macro and include `#include <zephyr/dt-bindings/display/panel.h>` at the top of file.

**Review by mniestroj — CHANGES_REQUESTED** [review]

File: boards/shields/seeed_xiao_round_display/seeed_xiao_round_display.overlay (hunk 7 above)
Comment (nicogou) [reply boards/shields/seeed_xiao_round_display/seeed_xiao_round_display.overlay:14]: I've got a Xiao ESP32S3 laying around somewhere, I'll try to do some testing with it (hopefully this weekend)

File: boards/shields/seeed_xiao_round_display/seeed_xiao_round_display.overlay [hunk 15]
```diff
@@ -0,0 +1,87 @@
...
+	pcf8563_xiao_round_display: pcf8563@51 {
+		status = "okay";
+		compatible = "nxp,pcf8563";
+		reg = <0x51 >;
+	};
+
+	chsc6x_xiao_round_display: chsc6x@2e {
+		status = "okay";
+		compatible = "chipsemi,chsc6x";
+		reg = < 0x2e >;
```
Comment (mniestroj) [inline boards/shields/seeed_xiao_round_display/seeed_xiao_round_display.overlay:56]: Nit: please remove spaces around register value and `<>`.

File: boards/shields/seeed_xiao_round_display/seeed_xiao_round_display.overlay [hunk 16]
```diff
@@ -0,0 +1,87 @@
...
+		compatible = "zephyr,lvgl-pointer-input";
+		input = <&chsc6x_xiao_round_display>;
+	};
+
+	aliases {
+		rtc = &pcf8563_xiao_round_display;
+	};
+};
+
+&xiao_adc{
```
Comment (mniestroj) [inline boards/shields/seeed_xiao_round_display/seeed_xiao_round_display.overlay:32]:
```suggestion
&xiao_adc {
```

File: boards/seeed/seeeduino_xiao/seeed_xiao_connector.dtsi [hunk 17]
```diff
@@ -30,3 +30,4 @@ xiao_spi: &sercom0 {};
 xiao_i2c: &sercom2 {};
 xiao_serial: &sercom4 {};
 xiao_dac: &dac0 {};
+xiao_adc: &adc {};
```
Comment (mniestroj) [inline boards/seeed/seeeduino_xiao/seeed_xiao_connector.dtsi:33]: You can remove:
> Needs the modification in PR https://github.com/zephyrproject-rtos/zephyr/pull/70683 in order to be able to use the
gcx01x driver without the reset-gpios property.

from the commit message and replace it with some information about this `xiao_adc` nodelabel. Or introduce this as part of separate commiit.

**Review by mniestroj — COMMENTED** [review]
Just minor things, otherwise LGTM

File: boards/seeed/seeeduino_xiao/seeed_xiao_connector.dtsi (hunk 17 above)
Comment (nicogou) [reply boards/seeed/seeeduino_xiao/seeed_xiao_connector.dtsi:33]: I did the requested changes, thanks for the reviews :)

**Review by mniestroj — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Review by kartben — APPROVED** [review]
Thanks for the great work!

**Review by danieldegrasse — APPROVED** [review]
